src/block-components/progress-bar/style.js

```diff
@@ -87,6 +87,16 @@ const getStyleParams = ( { isCircle } ) => {
 			attrName: 'progressBorderRadius',
 			hasUnits: 'px',
 		},
+		{
+			selector,
+			styleRule: '--progress-bar-border-radius',
+			attrName: 'progressApplyBarRadius',
+			valuePreCallback: ( value, getAttribute ) => {
+				const borderRadius = getAttribute( 'progressBorderRadius' )
+				return value ? borderRadius : undefined
+			},
+			hasUnits: 'px',
```

Comment:
2 things:
1. `hasUnits` isn't needed here, if you need to format the value, you can do `format: '%spx'`
2. We'll need `dependencies: [ 'progressApplyBarRadius' ]` here